### PR TITLE
FM-667 Supplier response to contract offer

### PR DESCRIPTION
**Pull request by tim-s-ccs**
- Created page for supplier response to contract offer
- Add validation logic for page
- Added the validation error messages
- Add JS for the character count due to form error issue
- Added correct routing for the pages
- Added the state changes
- Updated heading to be correct and links
- Translations and code quality

Still need to do gov notify

### CODE REVIEW

File: app/models/facilities_management/procurement_supplier.rb [hunk 1]
```diff
@@ -5,14 +5,20 @@ class ProcurementSupplier < ApplicationRecord
     default_scope { order(direct_award_value: :asc) }
     belongs_to :procurement, class_name: 'FacilitiesManagement::Procurement', foreign_key: :facilities_management_procurement_id, inverse_of: :procurement_suppliers
 
+    attribute :contract_response
+
+    before_validation :convert_to_boolean, on: :contract_response
+    validates :contract_response, inclusion: { in: [true, false] }, on: :contract_response
+    validates :reason_for_closing, presence: true, length: 1..500, if: :contract_response_false?, on: :contract_response
```
Comment (iceraluk) [inline app/models/facilities_management/procurement_supplier.rb]: might be useful some unit test for the validations

File: app/models/facilities_management/procurement_supplier.rb [hunk 2]
```diff
@@ -5,14 +5,20 @@ class ProcurementSupplier < ApplicationRecord
...
     aasm do
       state :unsent, initial: true
       state :sent, before_enter: %i[assign_contract_number set_date_and_send_email]
-      state :accepted, before_enter: %i[set_date_and_send_supplier_response]
+      state :accepted, before_enter: %i[set_supplier_response_date send_accepted_email]
       state :signed
       state :not_signed
       state :withdrawn
-      state :declined
+      state :declined, before_enter: %i[set_supplier_response_date send_declined_email]
```
Comment (iceraluk) [inline app/models/facilities_management/procurement_supplier.rb]: might be useful some unit tests for the state change

Comment (tim-s-ccs) [reply app/models/facilities_management/procurement_supplier.rb]: Unit tests added as requested

File: app/models/facilities_management/procurement_supplier.rb (hunk 1 above)
Comment (tim-s-ccs) [reply app/models/facilities_management/procurement_supplier.rb]: Unit tests added as requested

**Pull request merged**